frontend/src/components/AuthProvider.tsx

```diff
@@ -0,0 +1,44 @@
+// SPDX-FileCopyrightText: 2024 German Aerospace Center (DLR) and CISPA Helmholtz Center for Information Security
+// SPDX-License-Identifier: Apache-2.0
+
+import React from 'react';
+import {ReactNode} from 'react';
+import {AuthProvider as OAuth2WithPkceProvider, TAuthConfig} from 'react-oauth2-code-pkce';
+import {useAppSelector} from 'store/hooks';
+
+interface AuthProviderProps {
+  children: ReactNode;
+}
+
+function AuthProvider({children}: AuthProviderProps) {
+  const realm = useAppSelector((state) => state.realm.name);
+  let authConfig: TAuthConfig;
+
+  if (!import.meta.env.VITE_OAUTH_CLIENT_ID || !import.meta.env.VITE_OAUTH_API_URL) {
+    // in case required auth env vars are not set
+    console.warn(
+      'Missing environment variables: VITE_OAUTH_CLIENT_ID or VITE_OAUTH_API_URL. Please set it to enable authentication.'
+    );
+    authConfig = {
+      clientId: 'client-placeholder',
+      authorizationEndpoint: 'auth-endpoint-placeholder',
+      tokenEndpoint: 'token-endpoint-placeholder',
+      redirectUri: window.location.origin,
```

Comment:
This should be href, because it would break the redirect if ESID is not hosted at the root.

```suggestion
      redirectUri: window.location.href,
```